sonar-python-plugin/src/test/java/org/sonar/plugins/python/SonarLintPythonIndexerTest.java

```diff
@@ -0,0 +1,185 @@
+    assertThat(projectLevelSymbolTable.getSymbolsFromModule("mod")).isNull();
+    pythonIndexer.process(moduleFileEvent);
+    assertThat(projectLevelSymbolTable.getSymbolsFromModule("mod")).isNull();
+
+  }
+
+  @Test
+  public void test_indexer_added_file() throws IOException {
+    InputFile file1 = inputFile("main.py");
+    InputFile file2 = inputFile("mod.py");
+    SonarLintPythonIndexer pythonIndexer = new SonarLintPythonIndexer(new TestModuleFileSystem(Arrays.asList(file1, file2)));
+    pythonIndexer.buildOnce(context);
+    ProjectLevelSymbolTable projectLevelSymbolTable = pythonIndexer.projectLevelSymbolTable();
+
+    InputFile file3 = createInputFile("added.py");
+    ModuleFileEvent moduleFileEvent = mock(ModuleFileEvent.class);
+    when(moduleFileEvent.getType()).thenReturn(ModuleFileEvent.Type.CREATED);
+    when(moduleFileEvent.getTarget()).thenReturn(file3);
+    pythonIndexer.process(moduleFileEvent);
+    assertThat(projectLevelSymbolTable.getSymbolsFromModule("main")).hasSize(1);
+    assertThat(projectLevelSymbolTable.getSymbolsFromModule("added")).hasSize(1);
+    Symbol newFuncSymbol = projectLevelSymbolTable.getSymbol("added.new_func");
+    assertThat(newFuncSymbol).isNotNull();
+    assertThat(newFuncSymbol.is(Symbol.Kind.FUNCTION)).isTrue();
+  }
+
+  @Test
+  public void test_indexer_added_nonexistent_file() throws IOException {
+    InputFile file1 = inputFile("main.py");
+    InputFile file2 = inputFile("mod.py");
+    SonarLintPythonIndexer pythonIndexer = new SonarLintPythonIndexer(new TestModuleFileSystem(Arrays.asList(file1, file2)));
+    pythonIndexer.buildOnce(context);
+    ProjectLevelSymbolTable projectLevelSymbolTable = pythonIndexer.projectLevelSymbolTable();
+
+    InputFile nonExistentFile = TestInputFileBuilder.create("moduleKey", "nonexistent.py")
+      .setModuleBaseDir(baseDir.toPath())
+      .setCharset(StandardCharsets.UTF_8)
+      .setType(InputFile.Type.MAIN)
+      .setLanguage(Python.KEY)
+      .build();
+    ModuleFileEvent moduleFileEvent = mock(ModuleFileEvent.class);
+    when(moduleFileEvent.getType()).thenReturn(ModuleFileEvent.Type.CREATED);
+    when(moduleFileEvent.getTarget()).thenReturn(nonExistentFile);
+    pythonIndexer.process(moduleFileEvent);
+    assertThat(projectLevelSymbolTable.getSymbolsFromModule("main")).hasSize(1);
+    assertThat(projectLevelSymbolTable.getSymbolsFromModule("nonexistent")).isNull();
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).contains("Failed to load file \"nonexistent.py\" to the project symbol table");
+  }
+
+  @Test
+  public void test_indexer_modified_file() throws IOException {
+    InputFile file1 = inputFile("main.py");
+    InputFile file2 = inputFile("mod.py");
+    SonarLintPythonIndexer pythonIndexer = new SonarLintPythonIndexer(new TestModuleFileSystem(Arrays.asList(file1, file2)));
+    pythonIndexer.buildOnce(context);
+    ProjectLevelSymbolTable projectLevelSymbolTable = pythonIndexer.projectLevelSymbolTable();
```

Comment:
I would suggest declaring these vars as fields:
```java
  InputFile file1;
  InputFile file2;
  SonarLintPythonIndexer pythonIndexer;
  ProjectLevelSymbolTable projectLevelSymbolTable;
```

and move this initialization in the `@Before init()` method